cmd/pitr/storage/storage.go

```diff
@@ -0,0 +1,69 @@
+package storage
+
+import (
+	"context"
+	"io"
+
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	"github.com/pkg/errors"
+)
+
+// S3 is a type for working with S3 storages
+type S3 struct {
+	minioClient *minio.Client   // minio client for work with storage
+	ctx         context.Context // context for client operations
+	bucketName  string          // S3 bucket name where binlogs will be stored
+}
+
+// NewS3 return new Manager, useSSL using ssl for connection with storage
+func NewS3(endpoint, accessKeyID, secretAccessKey, bucketName, region string, useSSL bool) (*S3, error) {
+	minioClient, err := minio.New(endpoint, &minio.Options{
+		Creds:  credentials.NewStaticV2(accessKeyID, secretAccessKey, ""),
+		Secure: useSSL,
+		Region: region,
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "new minio client")
+	}
+
+	return &S3{
+		minioClient: minioClient,
+		ctx:         context.TODO(),
+		bucketName:  bucketName,
+	}, nil
+}
+
+// GetObject return content by given object name
+func (s *S3) GetObject(objectName string) (io.Reader, error) {
+	oldObj, err := s.minioClient.GetObject(s.ctx, s.bucketName, objectName, minio.GetObjectOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "get object")
+	}
+
+	return oldObj, nil
+}
+
+// PutObject puts new object to storage with given name and content
+func (s *S3) PutObject(name string, data io.Reader) error {
+	_, err := s.minioClient.PutObject(s.ctx, s.bucketName, name, data, -1, minio.PutObjectOptions{})
+	if err != nil {
+		return errors.Wrap(err, "put object")
+	}
+
+	return nil
+}
+
+func (s *S3) ListObjects(prefix string) []string {
+	opts := minio.ListObjectsOptions{
+		UseV1:  true,
+		Prefix: prefix,
+	}
+	list := []string{}
+
+	for object := range s.minioClient.ListObjects(s.ctx, s.bucketName, opts) {
+		list = append(list, object.Key)
+	}
```

Comment:
it is important to return results in `reverse-alphabetical` order, to reduce the amount of files which are downloaded by `setBinlogs` function

Reply:
we reverse it outsido of this method https://github.com/percona/percona-xtradb-cluster-operator/blob/K8SPXC-530/cmd/pitr/recoverer/recoverer.go#L243